iree/compiler/Dialect/HAL/Target/LLVM/internal/WasmLinkerTool.cpp

```diff
@@ -57,22 +57,16 @@ class WasmLinkerTool : public LinkerTool {
     return "";
   }
 
-  LogicalResult configureModule(llvm::Module *llvmModule,
-                                ArrayRef<StringRef> entryPointNames) override {
+  LogicalResult configureModule(
+      llvm::Module *llvmModule,
+      ArrayRef<llvm::Function *> exportedFuncs) override {
     for (auto &func : *llvmModule) {
       // Enable frame pointers to ensure that stack unwinding works.
       func.addFnAttr("frame-pointer", "all");
 
       // -ffreestanding-like behavior.
       func.addFnAttr("no-builtins");
     }
-
-    // https://lld.llvm.org/WebAssembly.html#exports
-    for (auto entryPointName : entryPointNames) {
-      auto *entryPointFn = llvmModule->getFunction(entryPointName);
-      entryPointFn->addFnAttr("wasm-export-name", entryPointName);
-    }
```

Comment:
These `wasm-export-name` attributes were load bearing for the entry points to be accessible through the generated wasm modules. The other mechanisms at the link I included here should also work, but may need different plumbing.

After this change (I suspect - haven't bisected), I get empty `.wat` files:
```
(module
  (table $T0 1 1 funcref)
  (memory $memory 2)
  (global $g0 (mut i32) (i32.const 66560))
  (export "memory" (memory 0)))
```

Reply:
Fix in https://github.com/google/iree/pull/5286